src/pages/iou/request/step/IOURequestStepCategory.js

```diff
@@ -50,11 +53,22 @@ function IOURequestStepCategory({
      * @param {String} category.searchText
      */
     const updateCategory = (category) => {
-        if (category.searchText === transaction.category) {
-            IOU.resetMoneyRequestCategory_temporaryForRefactor(transactionID);
-        } else {
-            IOU.setMoneyRequestCategory_temporaryForRefactor(transactionID, category.searchText);
+        const isSelectedCategory = category.searchText === transaction.category;
+        const updatedCategory = isSelectedCategory ? '' : category.searchText;
+
+        // The case edit split bill
+        if (isSplitBill && isEditing) {
+            IOU.setDraftSplitTransaction(transaction.transactionID, {category: category.searchText});
+            navigateBack();
+            return;
+        }
+        // The casse edit request
+        if (isEditing) {
+            IOU.updateMoneyRequestCategory(transaction.transactionID, report.reportID, updatedCategory);
+            Navigation.dismissModal();
+            return;
         }
+        IOU.setMoneyRequestCategory(transactionID, updatedCategory);
```

Comment:
I think we also should call setMoneyRequestCategory in the editing request flow to handle offline case. You can check [this PR](https://github.com/Expensify/App/pull/35137) for similar case